cordova-lib/src/cordova/platform.js

```diff
@@ -363,6 +363,7 @@ function remove(hooksRunner, projectRoot, targets, opts) {
             platformMetadata.remove(projectRoot, target);
         });
     }).then(function() {
+        opts.cordova.platforms = cordova_util.listPlatforms(opts.projectRoot);
```

Comment:
@vladimir-kotikov probably the root reason of the issue is in hooksRunner.opts.platforms [being set in platforms.js](https://github.com/apache/cordova-lib/blob/master/cordova-lib/src/cordova/platform.js#L570) and then used in [refineOpts in HooksRunner](https://github.com/apache/cordova-lib/blob/master/cordova-lib/src/hooks/HooksRunner.js#L69) - we will need to refine the way context.opts.cordova.platforms is used and properly document it.